Objects/descrobject.c

```diff
@@ -1637,6 +1609,50 @@ property_clear(PyObject *self)
     return 0;
 }
 
+#include "clinic/descrobject.c.h"
+
+PyTypeObject PyDictProxy_Type = {
```

Comment:
Since you're refactoring this piece of code anyways, maybe we can start using designated initializers?